imap-proto/src/parser/core.rs

```diff
@@ -53,16 +58,27 @@ pub fn is_quoted_specials(c: u8) -> bool {
     c == b'"' || c == b'\\'
 }
 
-// literal = "{" number "}" CRLF *CHAR8
-//            ; Number represents the number of CHAR8s
-named!(pub literal<&[u8]>, do_parse!(
-    tag!("{") >>
-    len: number >>
-    tag!("}") >>
-    tag!("\r\n") >>
-    data: take!(len) >> // FIXME: 0x00 is not allowed
-    (data)
-));
+/// literal = "{" number "}" CRLF *CHAR8
```

Comment:
I'd like to split this into two commits: one commit that converts `literal` to function-style, and one commit that makes it more strict. Otherwise, it makes reviewing stuff much harder.